driver/src/orderbook/streamed/updating_orderbook.rs

```diff
@@ -0,0 +1,75 @@
+use super::*;
+use crate::contracts::stablex_contract::batch_exchange;
+use anyhow::{anyhow, bail, Result};
+use block_timestamp::BlockTimestamp;
+use ethcontract::{contract::Event, errors::ExecutionError};
+use futures::stream::{BoxStream, StreamExt as _};
+use orderbook::Orderbook;
+use std::future::Future;
+use std::sync::{mpsc, Arc, RwLock};
+
+// TODO:
+// `pub struct UpdatingOrderbook` that also owns an `Arc<RwLock<Orderbook>>` and creates a
+// background thread running `UpdatingOrderbookThread::update_with_events`.
+// This way `UpdatingOrderbookThread` is independently testable.
+
+/// Update the orderbook with events from the stream forever or until exit_indicator is dropped.
+///
+/// Returns Ok when exit_indicator is dropped.
+/// Returns Err if the stream ends.
+async fn update_with_events_forever(
+    orderbook: Arc<RwLock<Orderbook>>,
```

Comment:
nit: You will probably get better performance from a `Mutex` here seeing as there is only one reader.